tests/target/issue-4036.rs

```diff
@@ -0,0 +1,12 @@
+// rustfmt-format_strings: true
+
```

Comment:
Realize this wasn't in the report, but could you do me a favor and add another set of tests that have `hard_tabs` turned on? We've got some idempotency issues within macro defs in some similar looking cases, and I'd like to rule out those occurring here (or if they do, at least confirm it's nothing new/same behavior could be produced on master)

Reply:
I've added a test with `hard_tabs` enabled. Let me know if you would like the test modified or additional tests added.